src/bun.js/api/server/SSLConfig.zig

```diff
@@ -227,6 +227,89 @@ pub fn deinit(this: *SSLConfig) void {
         this.ca = null;
     }
 }
+pub fn clone(this: *const SSLConfig) SSLConfig {
+    var cloned: SSLConfig = .{
+        .secure_options = this.secure_options,
+        .request_cert = this.request_cert,
+        .reject_unauthorized = this.reject_unauthorized,
+        .client_renegotiation_limit = this.client_renegotiation_limit,
+        .client_renegotiation_window = this.client_renegotiation_window,
+        .requires_custom_request_ctx = this.requires_custom_request_ctx,
+        .is_using_default_ciphers = this.is_using_default_ciphers,
+        .low_memory_mode = this.low_memory_mode,
+        .protos_len = this.protos_len,
+    };
+    const fields = .{
+        "server_name",
+        "key_file_name",
+        "cert_file_name",
+        "ca_file_name",
+        "dh_params_file_name",
+        "passphrase",
+        "protos",
+    };
+
+    inline for (fields) |field| {
+        if (@field(this, field)) |slice_ptr| {
+            const slice = std.mem.span(slice_ptr);
+            @field(cloned, field) = bun.handleOom(bun.default_allocator.dupeZ(u8, slice));
+        }
+    }
```

Comment:
```suggestion
    inline for (fields_cloned_by_memcopy) |field_name| {
        const source_slice = std.mem.span(@field(this, field) orelse continue);
        @field(cloned, field) = bun.handleOom(bun.default_allocator.dupeZ(u8, slice));
    }
```

[I'm a never-nester](https://www.youtube.com/watch?v=CFRhGnuXG-4)

Reply:
I feel that in this case is harder to follow